### PR TITLE
enhance(build): logic parsing and linking of auto cancel messaging

### PR DESCRIPTION
**Pull request by ecrupper**
Types / Server PRs: https://github.com/go-vela/types/pull/299 & https://github.com/go-vela/server/pull/911

UI essentially determines whether a build was canceled or auto-canceled based on the content of the error message and then links the superseding build in the message itself.

Will update description with styling decisions when made and bring the PR out of draft at that point.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ services:
   # https://go-vela.github.io/docs/administration/server/
   server:
     container_name: server
-    image: target/vela-server:latest
+    image: server:local
```
Comment (wass3r) [inline docker-compose.yml:37]: suggestion: not sure if this was intended to be checked in, but i'd personally prefer to revert this. it assumes a certain work style. for example, if i want to pair local dev with certain server code, i temporarily point this at the local server repo to live build the image. leaving it as it was will typically allow folks to get the local stack running without having to follow extra steps.

**Review by wass3r — APPROVED** [review]
Feels like cheating but I happen to approve :) nice work.

**Review by claire1618 — APPROVED** [review]
seems good to me!

**Review by plyr4 — APPROVED** [review]
<3

**Pull request merged**